ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/PaywallResourceProvider.kt

```diff
@@ -56,6 +59,25 @@ internal class PaywallResourceProvider(
     override fun getResourceIdentifier(name: String, type: String): Int =
         resources.getIdentifier(name, type, packageName)
 
+    @Suppress("ReturnCount")
+    override fun getXmlFontFamily(resourceId: Int): FontFamily? {
+        val parser = try {
+            resources.getXml(resourceId)
+        } catch (_: Resources.NotFoundException) {
+            return null
+        }
+        return try {
+            FontFamilyXMLParser.parse(parser)
+        } catch (@Suppress("TooGenericExceptionCaught") e: Throwable) {
+            // This can happen if the XML is malformed or not a valid font family.
+            // We log the error and return null.
+            Logger.e("Error parsing XML font family with resource ID $resourceId", e)
```

Comment:
Printing the resource name is a bit more helpful, as the ID is basically random:
```suggestion
            Logger.e("Error parsing XML font family with resource ID ${resources.getResourceEntryNameOrNull(resourceId) ?: resourceId}", e)
```

with `getResourceEntryNameOrNull()` being: 
```kotlin
private fun Resources.getResourceEntryNameOrNull(resourceId: Int): String? =
    try {
        getResourceEntryName(resourceId)
    } catch (_: Resources.NotFoundException) {
        null
    }
```